app/src/main/java/org/schabi/newpipe/util/ServiceHelper.java

```diff
@@ -113,18 +117,32 @@ public static int getImportInstructionsHint(final int serviceId) {
     }
 
     public static int getSelectedServiceId(final Context context) {
+        return Optional.ofNullable(getSelectedService(context))
+                .orElse(DEFAULT_FALLBACK_SERVICE)
+                .getServiceId();
+    }
+
+    @Nullable
+    public static StreamingService getSelectedService(final Context context) {
         final String serviceName = PreferenceManager.getDefaultSharedPreferences(context)
                 .getString(context.getString(R.string.current_service_key),
                         context.getString(R.string.default_service_value));
 
-        int serviceId;
         try {
-            serviceId = NewPipe.getService(serviceName).getServiceId();
+            return NewPipe.getService(serviceName);
         } catch (final ExtractionException e) {
-            serviceId = DEFAULT_FALLBACK_SERVICE.getServiceId();
+            return null;
         }
+    }
 
-        return serviceId;
+    @Nonnull
+    public static String getNameOfServiceById(final int serviceId) {
+        return ServiceList.all().stream()
+                .filter(s -> s.getServiceId() == serviceId)
+                .findFirst()
+                .map(StreamingService::getServiceInfo)
+                .map(StreamingService.ServiceInfo::getName)
+                .orElse("<unknown>");
```

Comment:
String resource?

Reply:
Wouldn't recommend:
1. This should never happen in the first place (however it's possible because we use ``int``s instead of valid objects)
2. We are inside a static method we would need some kind of context here (and in all subsequent calls)
3. I don't think we have to translate this
4. It was a hardcoded string before